pkg/beacon/analyzer.go

```diff
@@ -295,46 +299,51 @@ func (a *analyzer) hostIcertQuery(icert bool, src data.UniqueIP, dst data.Unique
 	// create query
 	query := bson.M{}
 
-	// update host table if there is an invalid cert record between pair
-	if icert {
-
-		newFlag := false
+	// don't update host table if there isn't an invalid cert record between pair
+	if !icert {
+		return updateInfo{}
+	}
 
-		var resList []interface{}
+	hostSelector := src.BSONKey()
+	hostSelector["dat.icdst"] = dst.BSONKey()
```

Comment:
There is a bug here that is fixed in #683 and there will likely be merge conflicts.